### PR TITLE
Revise log

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - refactor ngraph log class
 - add type_prop tests
 - minor change in spec
### Tickets:
 - 37447

### CODE REVIEW

File: docs/ops/arithmetic/Log_1.md [hunk 1]
```diff
@@ -6,6 +6,12 @@
...
 
+**Detailed description**: *Log* does the following with the input tensor *a*:
+
+\f[
+a_{i} = log(a_{i})
+\f]
+
 **Attributes**:
 
     No attributes available.
```
Comment (jdanieck) [inline docs/ops/arithmetic/Log_1.md:17]: **Inputs**
* **1**: An tensor of type T and arbitrary shape. **Required.**  

**Outputs**
* **1**: The result of element-wise log operation. A tensor of type T and the same shape as input.

**Review by jdanieck — APPROVED** [review]

File: ngraph/test/type_prop/unary_ops.cpp [hunk 2]
```diff
@@ -96,6 +96,6 @@ REGISTER_TYPED_TEST_CASE_P(UnaryOperator,
                            dynamic_rank_input_shape_3D,
                            dynamic_rank_input_shape_full);
 
-using Types = ::testing::Types<op::Acos, op::Asin, op::Abs, op::Sqrt, op::Sin, op::Exp, op::Floor>;
+using Types = ::testing::Types<op::Acos, op::Asin, op::Abs, op::Sqrt, op::Sin, op::Exp, op::Floor, op::Log>;
```
Comment (ilyachur) [inline ngraph/test/type_prop/unary_ops.cpp:99]: Yesterday I have discussed with @pszmel that will be good to split `unary_ops` tests as we already done for `arithmetic_ops`. @pszmel did you create a ticket for it?

Comment (pszmel) [reply ngraph/test/type_prop/unary_ops.cpp:99]: @ilyachur Not yet, but I will create it today.

**Review by ilyachur — APPROVED** [review]

File: ngraph/test/type_prop/unary_ops.cpp (hunk 2 above)
Comment (jdanieck) [reply ngraph/test/type_prop/unary_ops.cpp:99]: Ticket: CVS-57678. It is scheduled for next sprint.

**Comment by jdanieck** [comment]
@ilyachur CI green, ready to merge.

**Pull request merged**